app/controllers/declaration/ConsignorEoriNumberController.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.navigation.Navigator
+import forms.DeclarationPage
+import forms.common.YesNoAnswer.YesNoAnswers
+import forms.declaration.{CarrierDetails, ExporterDetails}
+import forms.declaration.consignor.ConsignorEoriNumber.form
+import forms.declaration.consignor.{ConsignorDetails, ConsignorEoriNumber}
+import javax.inject.Inject
+import models.requests.JourneyRequest
+import models.{ExportsDeclaration, Mode}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.declaration.consignor_eori_number
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ConsignorEoriNumberController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  navigator: Navigator,
+  mcc: MessagesControllerComponents,
+  consignorEoriDetailsPage: consignor_eori_number,
+  override val exportsCacheService: ExportsCacheService
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable {
+
+  def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
```

Comment:
And this controller require the same for both methods.

It will be easier to understand in what journey this page is available

Reply:
This has been addressed.